### PR TITLE
Upgrade ffmpeg to 2.8.

### PR DESCRIPTION
**Pull request by wolfman2000**
This is a big leap. Some things may break. Hopefully nothing does.

### CODE REVIEW

**Comment by wolfman2000** [comment]
Current status:

* Linux: compiles and links.
* Mac OS X: compiles, but doesn't link.
* Windows: does not compile.

More work is needed.

**Comment by Prcuvu** [comment]
The Windows build is not actually 2.8.

Maybe it can be bundled in 5.1.0, along with major upgrades for all other external libraries?

**Comment by wolfman2000** [comment]
Wait...what? I saw the zip file I downloaded: it claimed 2.8.0. Which version did I get instead?

File: extern/ffmpeg/README.txt [hunk 1]
```diff
@@ -2,17 +2,15 @@ This is a FFmpeg Win32 shared build by Kyle Schwarz.
...
-FFmpeg version: 2.1.3
-  libavutil      52. 48.101 / 52. 48.101
-  libavcodec     55. 39.101 / 55. 39.101
-  libavformat    55. 19.104 / 55. 19.104
-  libavdevice    55.  5.100 / 55.  5.100
-  libavfilter     3. 90.100 /  3. 90.100
-  libswscale      2.  5.101 /  2.  5.101
-  libswresample   0. 17.104 /  0. 17.104
-  libpostproc    52.  3.100 / 52.  3.100
+FFmpeg version: 20151006-git-d1c8368
```
Comment (Prcuvu) [inline extern/ffmpeg/README.txt]: Well... Maybe Zeranoe's typo?

**Comment by wolfman2000** [comment]
@Prcuvu has a better handle than I did. Closing my attempt.

**Pull request closed without merging**